src/t_zset.c

```diff
@@ -2104,6 +2105,8 @@ inline static void zunionInterAggregate(double *target, double val, int aggregat
         *target = val < *target ? val : *target;
     } else if (aggregate == REDIS_AGGR_MAX) {
         *target = val > *target ? val : *target;
+    } else if (aggregate == REDIS_AGGR_MUL) {
+        *target = *target * val;
```

Comment:
Probably requires some NAN protection. Maybe something like:
```suggestion
        *target = *target * val;
        if (isnan(*target)) *target = 0.0;
```
Better verify my theory...